code/modules/europa/machinery/consoles/console.dm

```diff
@@ -7,16 +7,21 @@ var/console_count = 0
 	icon_state = "console"
 	is_data_console = 1
 
-	var/show_terminals = 1
-	var/show_machines = 1
-	var/show_sensors = 1
-
+	var/list/installed_software = list()
+	var/global/list/default_software = list(
+		/datum/console_module/configuration,
+		/datum/console_module/basic_interface,
+		/datum/console_module/sensor_data
+		)
 	var/on = 1
 
 /obj/machinery/europa/console/initialize()
 	..()
 	if(name == "terminal")
 		name = "terminal (#[++console_count])"
+	for(var/module_type in default_software)
+		var/datum/console_module/module = new module_type(src)
+		installed_software[module.name] = module
```

Comment:
You should loop over and qdel each installed software in Destroy(), and then Cut the list to make sure of a clean GC.